### PR TITLE
initial ogcapi-processes implementation

### PR DESCRIPTION
**Pull request by cehbrecht**
This PR is first implementation for ogcapi-processes (#749).

### CODE REVIEW

**Comment by cehbrecht** [comment]
Open issues:
* using/adding offline tests
* job status and cancel not tested
* handing POST and DELETE requests ... should go into base classes
* exception handling ... probably also part of base classes
* naming conventions `process_status` ...
* json inputs and outputs look simple ... but we might like to have something with python: https://pydantic-docs.helpmanual.io/datamodel_code_generator/
* server implementations are different ... not complete ... different responses ... POST blocked?

**Comment by huard** [comment]
FYI: https://github.com/opengeospatial/ogcapi-processes/pull/121/files

File: owslib/ogcapi/processes.py [hunk 1]
```diff
@@ -83,15 +100,28 @@ def execute(self, process_id: str, json: dict) -> dict:
         """
         # validate input
         m.Execute.parse_obj(json)
-
         path = f'processes/{process_id}/jobs'
         resp = self._request_post(path, json)
+
+        # Carsten, is it a server non-conformity that we can't just return m.Result.parse_obj(resp.json()) ?
```
Comment (cehbrecht) [inline owslib/ogcapi/processes.py:106]: I was unsure were the `location` parameter is. On weaver it is returned via the header ... but it should also be the json.

**Comment by tomkralidis** [comment]
Given there is an OAProc implementation now in master, should we close this?

**Comment by tomkralidis** [comment]
@cehbrecht can we remove this branch?

**Pull request closed without merging**